### PR TITLE
fix #285531: fix a crash on SVG export of some scores with MM rests

### PR DESCRIPTION
**Pull request by dmitrio95**
Fixes https://musescore.org/en/node/285531.

### CODE REVIEW

File: libmscore/system.cpp [hunk 1]
```diff
@@ -111,6 +111,7 @@ void System::clear()
 
 void System::appendMeasure(MeasureBase* mb)
       {
+      Q_ASSERT(!mb->isMeasure() || !toMeasure(mb)->hasMMRest());
```
Comment (mattmcclinch) [inline libmscore/system.cpp]: This Q_ASSERT fails when multimeasure rests are turned off. See [#285858: Turning off mmrests results in assertion failure](https://musescore.org/en/node/285858). What was the reason for making the assertion?

Comment (MarcSabatella) [reply libmscore/system.cpp]: I guess one might assume the measures being added to the system are either "real" measures with no mmrest covering them or else they are mmrest measures themselves - we'd never be adding a measure that is covered by an mmrest to the system.  That would seem a valid assumption to me as well, but apparently hasMMRest() is not a reliable way of checking this.  I think it returns true if the measure was *ever* covered by an mmrest, not just if it currently is.

Wouldn't be surprised if this turned out to be the source of other bugs as well - it's really kind of convoluted how we represent this.  But I think it's only valid to check hasMMRest() is mmresets are actually enabled.

Comment (dmitrio95) [reply libmscore/system.cpp]: @MarcSabatella is correct, this assertion is to ensure to some extent that we do not add measures that are covered by MM rests to the system. The assertion failure shows that this is not the correct form to check this so it should apparently be changed. Still that is good that it was discovered by this assertion failure: I used the same assumption in `System::nextMeasure()` in this patch to iterate over the system's measures, so this assertion failure implies that SVG export of certain scores is broken under the same conditions (after re-disabling MM rests). So both places should be changed. I will prepare a pull request to correct this.

Comment (dmitrio95) [reply libmscore/system.cpp]: See #4793, it looks like that is sufficient.

**Pull request merged**